processor/error/event.go

```diff
@@ -16,35 +17,96 @@ import (
 )
 
 type Event struct {
-	Id          *string
-	Culprit     *string
-	Context     common.MapStr
+	Id        *string
+	Culprit   *string
+	Context   common.MapStr
+	Timestamp time.Time
+
 	Exception   *Exception
 	Log         *Log
-	Timestamp   time.Time
-	Transaction *struct {
-		Id string
-	}
+	Transaction *Transaction
 
 	data common.MapStr
 }
 
+type Transaction struct {
+	Id string
+}
+
 type Exception struct {
-	Code       interface{}
 	Message    string
 	Module     *string
+	Code       interface{}
 	Attributes interface{}
-	Stacktrace m.Stacktrace `mapstructure:"stacktrace"`
+	Stacktrace m.Stacktrace
 	Type       *string
 	Handled    *bool
 }
 
 type Log struct {
-	Level        *string
 	Message      string
-	ParamMessage *string      `mapstructure:"param_message"`
-	LoggerName   *string      `mapstructure:"logger_name"`
-	Stacktrace   m.Stacktrace `mapstructure:"stacktrace"`
+	Level        *string
+	ParamMessage *string
+	LoggerName   *string
+	Stacktrace   m.Stacktrace
+}
+
+func DecodeEvent(input interface{}, err error) (*Event, error) {
+	if input == nil || err != nil {
+		return nil, err
+	}
+	raw, ok := input.(map[string]interface{})
+	if !ok {
+		return nil, errors.New("Invalid type for error event")
+	}
+	df := utility.DataFetcher{}
+	e := Event{
+		Id:        df.StringPtr(raw, "id"),
+		Culprit:   df.StringPtr(raw, "culprit"),
+		Context:   df.MapStr(raw, "context"),
+		Timestamp: df.TimeRFC3339(raw, "timestamp"),
+	}
+	transactionId := df.StringPtr(raw, "id", "transaction")
+	if transactionId != nil {
+		e.Transaction = &Transaction{Id: *transactionId}
+	}
+
+	var str *m.Stacktrace
+	err = df.Err
+	ex := df.MapStr(raw, "exception")
+	exMsg := df.StringPtr(ex, "message")
+	if exMsg != nil {
+		e.Exception = &Exception{
+			Message:    *exMsg,
+			Code:       df.Interface(ex, "code"),
+			Module:     df.StringPtr(ex, "module"),
+			Attributes: df.Interface(ex, "attributes"),
+			Type:       df.StringPtr(ex, "type"),
+			Handled:    df.BoolPtr(ex, "handled"),
+			Stacktrace: m.Stacktrace{},
+		}
+		str, err = m.DecodeStacktrace(ex["stacktrace"], err)
```

Comment:
nitpick: when i read `str = ...`  I thought why `DecodeStacktrace` would return a string... i'd rename this variable :)
(and same for the Log part bellow and Span)